crates/holochain/src/core/ribosome/host_fn/create.rs

```diff
@@ -193,6 +201,65 @@ pub mod wasm_test {
         assert_eq!(round_twice, vec![round.clone(), round],);
     }
 
+    #[test]
+    fn ribosome_create_entry_network_test() {
+        tokio::runtime::Builder::new_multi_thread()
+            // Need a bigger stack for this test for some reason.
+            .thread_stack_size(4_000_000)
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(async move {
+                // This spawn is required so that the thread size config takes
+                // effect because a block_on doesn't count as a worker.
+                let handle = tokio::spawn(async move {
+                holochain_trace::test_run().ok();
+
+                let mut network_topology = NetworkTopology::default();
+
+                let (dna_file, _, _) = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+
+                network_topology.add_dnas(vec![dna_file.clone()]);
+
+                let size_fact = SizedNetworkFact { nodes: 7, agents: 1..=3 };
+                let partition_fact = StrictlyPartitionedNetworkFact {
+                    partitions: 1,
+                    efficiency: 1.0,
+                };
+                let density_fact = DenseNetworkFact { density: 0.8 };
+
+                let mut facts = facts![size_fact, partition_fact, density_fact];
+
+                let mut g: Generator = unstructured_noise().into();
+                let mut rng = rng_from_generator(&mut g);
+                network_topology = facts.mutate(&mut g, network_topology).unwrap();
+
+                network_topology.apply().await.unwrap();
+
+                let alice_node = network_topology.random_node(&mut rng).unwrap();
+                let alice_cell = alice_node.cells().into_iter().filter(|cell| cell.dna_hash() == dna_file.dna_hash()).choose(&mut rng).unwrap();
+                let alice = alice_node.conductor().lock().await.read().await.get_sweet_cell(alice_cell).unwrap();
+
+                let bob_node = network_topology.random_node(&mut rng).unwrap();
+                let bob_cell = bob_node.cells().into_iter().filter(|cell| cell.dna_hash() == dna_file.dna_hash()).choose(&mut rng).unwrap();
+                let bob = bob_node.conductor().lock().await.read().await.get_sweet_cell(bob_cell).unwrap();
+
+                let action_hash: ActionHash = alice_node.conductor().lock().await.write().await.call(&alice.zome(TestWasm::Create), "create_entry", ()).await;
+
+                crate::wait_for_10s!(
```

Comment:
@jost-s when you have a network larger than just the writer and reader, the reader may ask someone other than the author for data when they do a get

as the network is randomly generated and we're selecting random nodes in the randomly generated network, there's no easy way to predict who the reader is or how they're going to be reading (e.g. which authority(s) they will ask)

we ask the reader to poll the network until they see what they expect to see, because it's going to take a non deterministic amount of time to propagate through validation and gossip etc.